app/src/main/java/com/android/bookswap/ui/profile/UserProfile.kt

```diff
@@ -0,0 +1,168 @@
+
+  var needRecompose by remember { mutableStateOf(false) }
+
+  if (showEditProfile) {
+    EditProfileDialog(
+        onDismiss = {
+          showEditProfile = false
+          needRecompose = true
+        },
+        onSave = {
+          userVM.updateUser(
+              greeting = it.greeting,
+              firstName = it.firstName,
+              lastName = it.lastName,
+              email = it.email,
+              phone = it.phoneNumber,
+              user.latitude,
+              user.longitude,
+              picURL = user.profilePictureUrl)
+          showEditProfile = false
+          needRecompose = true
+        },
+        dataUser = user,
+        modifier = Modifier.testTag("editProfileAld"))
+  }
+
+  LaunchedEffect(userVM.uuid, needRecompose) {
+    user = userVM.getUser()
+    needRecompose = false
+  }
+
+  BookSwapAppTheme() {
+
+    // Scaffold to provide basic UI structure with a top app bar
+    Scaffold(
+        modifier = Modifier.testTag("profileScreenContainer"),
+        topBar = {
+          TopAppBar(
+              modifier = Modifier.testTag("profileTopAppBar"),
+              // Title of the screen
+              title = { Text("Your Profile", modifier = Modifier.testTag("profileTitleTxt")) },
+              // Icon button for navigation (currently no action defined)
+              navigationIcon = {
+                IconButton(modifier = Modifier.size(32.dp), onClick = {}) {
+                  Icon(
+                      imageVector = Icons.AutoMirrored.Sharp.ArrowBack,
+                      contentDescription = "",
+                      modifier = Modifier.size(30.dp))
+                }
+              })
+        }) {
+          // Column layout to stack input fields vertically with spacing
+          Row(
+              modifier = Modifier.padding(it).consumeWindowInsets(it).fillMaxWidth(),
+              horizontalArrangement = Arrangement.spacedBy(5f.dp)) {
+                Column(modifier = Modifier.fillMaxWidth(0.25f)) {
+                  Box {
+                    IconButton(onClick = { /*TODO*/}, modifier = Modifier.aspectRatio(1f)) {
+                      Box(
+                          modifier =
+                              Modifier.padding(2.5f.dp)
+                                  .border(3.5f.dp, Color(0xFFA98467), CircleShape)) {
+                            Image(
+                                imageVector = Icons.Rounded.AccountCircle,
+                                contentDescription = "",
+                                modifier = Modifier.fillMaxSize().scale(1.2f).clipToBounds(),
+                                colorFilter = ColorFilter.tint(Color(0xFF6C584C)))
+                          }
+                      Box(
+                          modifier = Modifier.fillMaxSize().padding(0f.dp),
+                          contentAlignment = Alignment.TopEnd) {
+                            IconButton(onClick = { /*TODO*/}, modifier = Modifier) {
```

Comment:
You should put a description in comment when you are using a TODO, so we know in the future what you wanted.

Reply:
This is a preview used in Android Studio to make UI design easier and faster by being able to render the UI without launching the app on the emulator, however , there is no way to "test" this part so it is currently commented to avoid reducing line coverage